src/libraries/System.Private.CoreLib/src/System/Globalization/JapaneseCalendar.iOS.cs

```diff
@@ -0,0 +1,98 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace System.Globalization
+{
+    public partial class JapaneseCalendar : Calendar
+    {
+        private static EraInfo[]? GetJapaneseErasNative()
+        {
+            if (GlobalizationMode.Invariant)
+            {
+                return null;
+            }
+
+            Debug.Assert(!GlobalizationMode.UseNls);
+
+            string[]? eraNames;
+            eraNames = Interop.Globalization.GetCalendarInfoNative("ja-JP", CalendarId.JAPAN, CalendarDataType.EraNames).Split("||");
+            if (eraNames.Length == 0)
+            {
+                return null;
+            }
+
+            List<EraInfo> eras = new List<EraInfo>();
+            int lastMaxYear = GregorianCalendar.MaxYear;
+            int latestEra = Interop.Globalization.GetLatestJapaneseEraNative();
+
+            for (int i = latestEra; i >= 0; i--)
+            {
+                DateTime dt;
+                if (!GetJapaneseEraStartDateNative(i, out dt))
+                {
+                    return null;
+                }
+
+                if (dt < s_calendarMinValue)
```

Comment:
Not sure for what test case, logic is taken from ICU flow https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Globalization/JapaneseCalendar.Icu.cs#L43. Native code doesn't return incorrect data, returns same as ICU , that is why there is need to do apply same logic for native code also.

Reply:
I understand. Then the code in this file is the same as https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Globalization/JapaneseCalendar.Icu.cs
with changes in 4 lines:
- https://github.com/dotnet/runtime/blob/8c566d4ffe865c6cc311eecf92b075863f0f0944/src/libraries/System.Private.CoreLib/src/System/Globalization/JapaneseCalendar.Icu.cs#L23
- https://github.com/dotnet/runtime/blob/8c566d4ffe865c6cc311eecf92b075863f0f0944/src/libraries/System.Private.CoreLib/src/System/Globalization/JapaneseCalendar.Icu.cs#L31
- https://github.com/dotnet/runtime/blob/8c566d4ffe865c6cc311eecf92b075863f0f0944/src/libraries/System.Private.CoreLib/src/System/Globalization/JapaneseCalendar.Icu.cs#L53
- https://github.com/dotnet/runtime/blob/8c566d4ffe865c6cc311eecf92b075863f0f0944/src/libraries/System.Private.CoreLib/src/System/Globalization/JapaneseCalendar.Icu.cs#L104

It's not a lot of changes, we could use the Icu file then with preprocessor directives in these 4 places, so that we don't duplicate the code.

Reply:
yes, will update the calls to not have duplicates. Thanks